hg-git: 0.8.5 -> 0.8.10

marsam opened the pull request:
###### Motivation for this change

Update hg-git which fixes an issue with the mercurial version shipped in nixpkgs.

Closes #30154

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

marsam committed:
hg-git: 0.8.5 -> 0.8.10

grahamc commented:
@GrahamcOfBorg build python2Packages.hg-git python3Packages.hg-git

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```

----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
builder for '/nix/store/g4f8vm1d9k6cy8xlbza9b77bap2qw73j-python3.6-hg-git-0.8.10.drv' failed with exit code 1
error: build of '/nix/store/g4f8vm1d9k6cy8xlbza9b77bap2qw73j-python3.6-hg-git-0.8.10.drv' failed on ‘mac1’
error: build of ‘/nix/store/g4f8vm1d9k6cy8xlbza9b77bap2qw73j-python3.6-hg-git-0.8.10.drv’ failed
```

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-linux

```


----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
builder for ‘/nix/store/ppfbd54n68ql88dllxvkphafwkz96a5b-python3.6-hg-git-0.8.10.drv’ failed with exit code 1
error: build of ‘/nix/store/ppfbd54n68ql88dllxvkphafwkz96a5b-python3.6-hg-git-0.8.10.drv’ failed
```

orivej committed:
hg-git: disable with python3

It is not importable from Python 3 and it is a plugin for mercurial, which is a
Python 2 application.

orivej commented:
Thank you!
